### ISSUE TITLE
closed port causes SEGV

### ISSUE THREAD

**Issue by ktakashi** [ktakashi/sagittarius-scheme#13]
**[Original report](https://bitbucket.org/ktakashi/sagittarius-scheme/issues/8) by Takashi Kato (Bitbucket: [ktakashi](https://bitbucket.org/ktakashi), GitHub: [ktakashi](https://github.com/ktakashi)).**

----------------------------------------

This script should print ok.
```
#!scheme
(import (scheme base)(scheme file))
((call-with-input-file "test.scm"
   (lambda(p)
     (lambda()
       (do ((i 0 (+ i 1)))
	   ((= i 10))
	 (read-char p))))))
(print 'ok)
```

**Comment by ktakashi** [comment]
**Original comment by Takashi Kato (Bitbucket: [ktakashi](https://bitbucket.org/ktakashi), GitHub: [ktakashi](https://github.com/ktakashi)).**

----------------------------------------

Fixed by 91e949b

**Comment by ktakashi** [comment]
**Original comment by Takashi Kato (Bitbucket: [ktakashi](https://bitbucket.org/ktakashi), GitHub: [ktakashi](https://github.com/ktakashi)).**

----------------------------------------

Added closed port check to read-char, peek-char, read and write-char.
Fixes #8
